include/meshio/obj.hpp

```diff
@@ -0,0 +1,284 @@
+
+            if (not mOBJName.compare(pOBJAttribute.mOBJName)) {
+                // Checking equality for mMaterialIDs.
+                if (mMaterialIDs.size() && pOBJAttribute.mMaterialIDs.size()) {
+                    for (unsigned i = 0; i < mMaterialIDs.size(); ++i) {
+                        if (not (mMaterialIDs[i] == pOBJAttribute.mMaterialIDs[i])) {
+                            return false;
+                        }
+                    }
+                    return true;
+                } else if (not mMaterialIDs.size() &&
+                           not pOBJAttribute.mMaterialIDs.size()) {
+                    return true;
+                } else {
+                    return false;
+                }
+
+            } else {
+                return false;
+            }
+
+        } else {
+            return false;
+        }
+    }
+
+};
+
+template<class T>
+class OBJData {
+  public:
+    std::vector< OBJAttribute<T> > mOBJAttributes;
+    std::vector<tinyobj::material_t> mOBJMaterials;
+
+    OBJData() {}
+    ~OBJData() {
+        clear();
+    }
+
+    // TODO: Check whether this can be made virtual by making OBJData derived
+    void print() {
+        for (unsigned i = 0; i < mOBJAttributes.size(); ++i) {
+            OBJAttribute<T> &objAttribute = mOBJAttributes[i];
+            std::cout << "Number of points = "
+                << objAttribute.mPositions.size() << std::endl;
+            for (unsigned j = 0; j < objAttribute.mPositions.size();
+                 ++j) {
+                std::cout << "v " << objAttribute.mPositions[j].getX() << " "
+                    << objAttribute.mPositions[j].getY() << " "
+                    << objAttribute.mPositions[j].getZ() << std::endl;
+            }
+
+            std::cout << "Number of indices = "
+                << objAttribute.mTriVertexIndices.size() << std::endl;
+            for (unsigned j = 0; j < objAttribute.mTriVertexIndices.size();
+                 ++j) {
+                std::cout << "f " << objAttribute.mTriVertexIndices[j].getX()
+                    << " " << objAttribute.mTriVertexIndices[j].getY()
+                    << " " << objAttribute.mTriVertexIndices[j].getZ()
+                    << std::endl;
+            }
+        }
+    }
+
+    void clear() {
+        for (unsigned i = 0; i < mOBJAttributes.size(); ++i) {
+            mOBJAttributes[i].clear();
+        }
+        mOBJAttributes.clear();
+        mOBJMaterials.clear();
+    }
+
+    bool isEqualOBJAttributes(const OBJData<T>& pOBJData) {
+        if (mOBJAttributes.size() == pOBJData.mOBJAttributes.size()) {
+            for (unsigned i = 0; i < mOBJAttributes.size(); ++i) {
+                if (not (mOBJAttributes[i] == pOBJData.mOBJAttributes[i])) {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    bool isEqualOBJMaterials(const OBJData<T>& pOBJData) {
```

Comment:
May be a name like `areMaterialsSame` can give more readable code ? `OBJ` doesn't need to be part of this name I think because this is method of OBJ related class object.